### PR TITLE
[Bugfix] Correct the loss function in RGCN model

### PR DESCRIPTION
**Pull request by yzh119**
## Description
As mentioned in #1204 , the loss function in our RGCN implementation is problematic. This PR fix the issue.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

File: examples/pytorch/rgcn/entity_classify.py [hunk 1]
```diff
@@ -117,8 +117,7 @@ def main(args):
         backward_time.append(t2 - t1)
         print("Epoch {:05d} | Train Forward Time(s) {:.4f} | Backward Time(s) {:.4f}".
               format(epoch, forward_time[-1], backward_time[-1]))
-        train_acc = torch.sum(logits[train_idx].argmax(dim=1) == labels[train_idx]).item() / len(train_idx)
-        val_loss = F.cross_entropy(logits[val_idx], labels[val_idx])
+        cross_entropy(logits[val_idx], labels[val_idx])
```
Comment (jermainewang) [inline examples/pytorch/rgcn/entity_classify.py]: `F.cross_entropy`? Also, isn't `tran_acc` required in L123?

Comment (yzh119) [reply examples/pytorch/rgcn/entity_classify.py]: Fixed.

**Comment by jermainewang** [comment]
Would you mind also fixing [the tensorflow example](https://github.com/dmlc/dgl/blob/master/examples/tensorflow/rgcn/entity_classify.py#L40) too?

**Comment by yzh119** [comment]
The accuracy become worse, we need to tune the hyper parameters again...
* AIFB: 94.44
* MUTAG: 72.06
* BGS: 75.86

**Comment by jermainewang** [comment]
@pbloem We are trying to fix the double-softmax problem. However, we found that the accuracy drops quite significantly. Have you observed the same issue?

**Comment by VoVAllen** [comment]
I already fixed it in tensorflow at https://github.com/dmlc/dgl/blob/master/examples/tensorflow/rgcn/entity_classify.py#L109

**Comment by yzh119** [comment]
@VoVAllen have you tuned hyper-parameters accordingly?

**Comment by VoVAllen** [comment]
@yzh119 It's unstable but the highest accuracy could reach the original result.

**Comment by pbloem** [comment]
@jermainewang Nope. I get the following mean accuracies over 10 repeats (standard deviation in brackets):

aifb: 0.961 (0.0194)
bgs: 0.834 (0.0317)
am: 0.892 (0.0167)

BGS is a little below what we had in the original paper, but the variance is high, so I think we just got lucky. All hyperparameters are the same as in the paper, and the weight initialization is taken from your code (ie. uniform xavier with relu gain everywhere).

You can find [my implementation here](https://github.com/pbloem/gated-rgcn/blob/3ec103c11048a8d31e555f5af108d3820dd7f8ee/kgmodels/models.py#L9), and the [training loop here](https://github.com/pbloem/gated-rgcn/blob/3ec103c11048a8d31e555f5af108d3820dd7f8ee/experiments/train.full.py#L100).

I strongly recommend averaging over 10 repeats, since the test sets are so small, but none of my runs on BGS are as low as .75.

**Comment by yzh119** [comment]
@pbloem thanks for your instructions!
